### PR TITLE
native, ui: move from TabStack to new custom NavBar component

### PR DESCRIPTION
**Pull request by jamesacklin**
Renovates the app navigation to use nested Native Stacks and a custom NavBar component.

- Rather than using createBottomTabNavigator, uses createNativeStackNavigator for all nested "home" screens (ChatList, Channel, Post, ChannelSearch).
- Adds a custom navigation component (NavBar) in packages/ui which seeks to be as generic as possible
- Adds a NavBarView to the app screens themselves, which in turn takes the navigation prop from the parent screen and handles all the navigate() calls itself

Functionality is exactly the same as before but eliminates the gray box on view transition (fixes LAND-1873). Thanks to @patosullivan for the talk-through assist.

Note that this is an absolutely-positioned element, so any screen that uses a scroller that extends below the bottom edge of the screen will need to be padded up at the bottom.

### CODE REVIEW

File: packages/ui/src/components/ChatList.tsx [hunk 1]
```diff
@@ -41,6 +41,7 @@ export function ChatList({
       gap: '$s',
       paddingTop: '$l',
       paddingHorizontal: '$l',
+      paddingBottom: '$6xl',
```
Comment (dnbrwstr) [inline packages/ui/src/components/ChatList.tsx:44]: Will this value be correct across different sized safe areas? And/or if we tweak navbar size? Would prefer to derive both from a `navBarHeight` constant + `insets.bottom` if possible.

File: packages/ui/src/components/NavBar/NavIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  id: string;
+  focused: boolean;
+  contact: Contact;
+  isLoading: boolean;
+  onPress?: () => void;
+}) {
+  return isLoading && !contact ? null : (
+    <View flex={1} onPress={onPress} alignItems="center" paddingTop={'$s'}>
+      <Avatar
+        width={22}
```
Comment (dnbrwstr) [inline packages/ui/src/components/NavBar/NavIcon.tsx:24]: Looks like ochre is size `22`, radius `4`?

File: packages/ui/src/components/ProfileScreenView.tsx [hunk 3]
```diff
@@ -32,7 +32,12 @@ export function Wrapped(props: Props) {
 
   return (
     <ScrollView>
-      <YStack flex={1} paddingHorizontal="$xl" paddingTop={top}>
+      <YStack
+        flex={1}
+        paddingHorizontal="$xl"
+        paddingTop={top}
+        paddingBottom={'$6xl'}
```
Comment (dnbrwstr) [inline packages/ui/src/components/ProfileScreenView.tsx:39]: Same comment re: bottom padding.

File: packages/ui/src/tamagui.config.ts [hunk 4]
```diff
@@ -54,6 +54,8 @@ export const tokens = createTokens({
     '2xl': 24,
     '3xl': 32,
     '4xl': 48,
+    '5xl': 64,
```
Comment (dnbrwstr) [inline packages/ui/src/tamagui.config.ts:57]: Intentional? Not seeing these used.

**Review by dnbrwstr — APPROVED** [review]
Looks great! A few nitpicks.

File: packages/ui/src/components/NavBar/NavIcon.tsx (hunk 2 above)
Comment (dnbrwstr) [reply packages/ui/src/components/NavBar/NavIcon.tsx:24]: `20`** oops

File: packages/ui/src/components/ChatList.tsx (hunk 1 above)
Comment (jamesacklin) [reply packages/ui/src/components/ChatList.tsx:44]: Makes sense, will do

**Review by latter-bolden — APPROVED** [review]

**Pull request merged**